scripts/us_census/pep/monthly_population/preprocess.py

```diff
@@ -0,0 +1,384 @@
+""" A Script to process USA Census PEP monthly population data
+    from the datasets in provided local path.
+    Typical usage:
+    1. python3 preprocess.py
+    2. python3 preprocess.py -i input_data
+"""
+import os
+import re
+import argparse
+
+import numpy as np
+import pandas as pd
+
+pd.set_option("display.max_columns", None)
+
+
+def extract_year(val):
+    """
+    This Methods returns true,year from the value contains year.
+    Otherwise false,''
+    val - Contains yyyy or yyyy [1] or .MM 1
+    """
+    val = str(val).strip().split(' ', maxsplit=1)[0]
+    if val.isnumeric() and len(val) == 4:
+        return True, val
+    return False, ''
+
+
+def return_year(col):
+    """
+    This Methods returns year value if col contains year.
+    Otherwise pandas NA value.
+    val - Contains yyyy or yyyy [1] or .MM 1
+    """
+    res, out = extract_year(col)
+    if res:
+        return out
+    return pd.NA
+
+
+def return_month(col):
+    """
+    This Methods returns month and date value if col contains month, date.
+    Otherwise pandas NA value.
+    val - Contains yyyy or yyyy [1] or .MM dd
+    """
+    res = extract_year(col)
+    if res[0]:
+        return pd.NA
+    return col
+
+
+def sum_cols(col):
+    """
+    This method adds Dataframe columns Year and Month, Date.
+    Returns the column value in format yyyymmdd. Example: 2022March01
+    """
+    if col[1] is None:
+        return col[0]
+    return col[0] + ' ' + col[1]
+
+
+def find_keep_value(curr_df: pd.DataFrame, new_df: pd.DataFrame):
+    """
+    This method returns keep value first or last while dropping
+    duplicates in the df. As few years repeating twice in few
+    datasets.
+    """
+    curr_max_year = max(
+        pd.to_datetime(curr_df['Date'], errors='coerce').dt.year)
+    new_max_year = max(pd.to_datetime(new_df['Date'], errors='coerce').dt.year)
+    if curr_max_year > new_max_year:
+        return "first"
+    return "last"
+
+
+class CensusUSACountryPopulation:
+    """
+    CensusUSACountryPopulation class provides methods
+    to load the data into dataframes, process, cleans
+    dataframes and finally creates single cleaned csv
+    file.
+    Also provides methods to generate MCF and TMCF
+    Files using pre-defined templates.
+    """
+
+    def __init__(self, input_files, csv_file_path, mcf_file_path,
+                 tmcf_file_path):
+        self.input_files = input_files
+        self.cleaned_csv_file_path = csv_file_path
+        self.mcf_file_path = mcf_file_path
+        self.tmcf_file_path = tmcf_file_path
+        self.df = None
+        self.file_name = None
+        self.scaling_factor = 1
+
+    def _load_data(self, file):
+        """
+        This Methods loads the data into pandas Dataframe
+        using the provided file path and Returns the Dataframe.
+        """
+        df = ""

```

Comment:
`df = None` when its type is going to change into a DataFrame later?

Reply:
Yes It is going to change into Dataframe later. But the reason to define it with None value, is for better memory usage?